### ISSUE TITLE
Consul integration for reading policies - Feature Request

### ISSUE THREAD

**Issue by mstrYoda** [open-policy-agent/opa#2539]
Currently as far as I know we have 2 option to get policies.

1 - Read from file
2 - Pull via provided bundle api

In our company we have currently implementing a feature to our infrastructure using OPA. We would like to OPA had feature like reading policies directly from consul.

For example:

`docker run -p 8181:8181 openpolicyagent/opa run -s --consulAddr=127.0.0.1:8500 --consulPolicyPath=/opa/policies`

Basically what we want is that OPA can read policies from given consul key path.

Also we can support **watch** while reading consul:

`docker run -p 8181:8181 openpolicyagent/opa run -s -w --consulAddr=127.0.0.1:8500 --consulPolicyPath=/opa/policies`

We can use consul watch api to capture key value changes.

What is your opinion about this feature?

I'm volunteer to implement such a feature if accepted.

**Comment by tsandall** [comment]
@mstrYoda thanks for for filing an issue.

Can you share the reasoning behind why you need OPA to integrate directly with Consul? I'm also curious about what alternatives you considered. E.g., one that jumps to mind is to write a simple HTTP service that serves the Bundle API backed by Consul. This gives you quite a bit of freedom and requires no changes in OPA.

If you really need to integrate OPA directly with Consul the current recommendation would be to fork and implement a [custom plugin](https://www.openpolicyagent.org/docs/latest/extensions/#custom-plugins-for-opa-runtime). At a high-level this is similar to how the [opa-istio-plugin](https://github.com/open-policy-agent/opa-istio-plugin) works.

We're unlikely to accept a direct integration at this point in time. The biggest concern is the precedent. There are other folks that have integrated OPA with other systems like Kafka or ZooKeeper for policy distribution. If we added direct support for Consul it would open the door to all of these and it's unclear that we want that.

**Comment by mstrYoda** [comment]
@tsandall Thank you for your help we can close this issue 👍
